pkg/pillar/types/assignableadapters.go

```diff
@@ -183,16 +187,46 @@ func (ioType IoType) IsNet() bool {
 }
 
 // AddOrUpdateIoBundle - Add an Io bundle to AA. If the bundle already exists,
-//  it just updates it.
+// the function updates it, while preserving the most specific information.
 func (aa *AssignableAdapters) AddOrUpdateIoBundle(ib IoBundle) {
 	curIbPtr := aa.LookupIoBundle(ib.Name)
 	if curIbPtr == nil {
-		log.Infof("handleIBCreate(%d %s %s) New Bundle",
+		log.Infof("AddOrUpdateIoBundle(%d %s %s) New bundle",
 			ib.Type, ib.Name, ib.AssignmentGroup)
 		aa.IoBundleList = append(aa.IoBundleList, ib)
-	} else {
-		*curIbPtr = ib
+		return
+	}
+	log.Infof("AddOrUpdateIoBundle(%d %s %s) Update bundle; diff %+v",
+		ib.Type, ib.Name, ib.AssignmentGroup,
+		cmp.Diff(*curIbPtr, ib))
+
+	// We preserve the most specific
+	if curIbPtr.UsedByUUID != nilUUID {
+		log.Infof("AddOrUpdateIoBundle(%d %s %s) preserve UsedByUUID %v",
+			ib.Type, ib.Name, ib.AssignmentGroup, curIbPtr.UsedByUUID)
+		ib.UsedByUUID = curIbPtr.UsedByUUID
+	}
+	if curIbPtr.IsPort {
```

Comment:
Thanks for fixing this.. 

I understand why we are preserving:
   curIbPtr.UsedByUUID
   curIbPtr.IsPCIBack

Why are we preserving:
   curIbPtr.MacAddr
   curIbPtr.IsPort

If these change - do we not support them?
If yes - should we add these checks earlier - instead of just logging them? The user should see these errors - right?

Reply:
What I noticed (in addition to the test failing) was that MacAddr and Unique was not set for all the Ethernet ports, which was odd. We actually use those fields to check against odd conditions when e.g., an Ethernet interface gets a different number when it comes back from pciback. So its somewhat important to not loose those values.

We have recorded UsedByUUID, IsPCIBack and IsPort based on information domainmgr received from nim. We shouldn't loose those just because we get some update from zedagent/the controller.